modelforge/train/tuning.py

```diff
@@ -0,0 +1,60 @@
+import torch
+from ray import tune, air
+from ray.tune.schedulers import ASHAScheduler
+
+
+def tune_model(
+    model: torch.nn.Module,
+    dataset: torch.utils.data.Dataset,
+    num_samples: int = 100,
+    name: str = "tune",
+):
+    """A function to tune a model.
+
+    Parameters
+    ----------
+    model : torch.nn.Module
+        The model to tune.
+
+    dataset : torch.utils.data.Dataset
+        The dataset to use for tuning.
+
+    num_samples : int, optional
+        The number of samples to use for tuning. Default is 100.
+
+    """
+    # access the model's configuration prior
+    # TODO: not finalized yet
+    config_prior = model._config_prior()
+
+    def objective():
+        raise NotImplementedError
+
+    scheduler = ASHAScheduler(
+        time_attr="training_iteration",
+        metric="rmse_vl",
+        mode="max",
+        max_t=100,
+        grace_period=10,
+        reduction_factor=3,
+        brackets=1,
+    )
+
+    tune_config = tune.TuneConfig(
+        scheduler=scheduler,
```

Comment:
Besides a `scheduler`, this also needs a `search_alg`, otherwise it's going to be random sampling. Consider perhaps `optuna`? 

https://docs.ray.io/en/latest/tune/examples/optuna_example.html